addons/account/models/account_reconcile_model.py

```diff
@@ -625,6 +626,23 @@ def _apply_rules(self, st_lines, excluded_ids=None, partner_map=None):
                     continue
 
                 excluded_lines_found = False
+                
+                # Add the lines already linked to the st_line if we only need to check an account move.
+                if self._context.get('mode') == 'to_check':
+                    query = '''
+                        SELECT aml.id FROM (
+                            SELECT aml.id, aml.full_reconcile_id
+                            FROM account_move_line aml
+                            JOIN account_move am ON aml.move_id = am.id
+                            WHERE am.to_check = True
+                            AND aml.full_reconcile_id > 0
+                            AND aml.id IN %s) as t
+                        JOIN account_move_line as aml ON aml.full_reconcile_id = t.full_reconcile_id AND aml.id != t.id;
```

Comment:
these conditions on full_reconcile_is are dubious... I don't get why it is not simply `line.journal_entry_ids.filtered(lambda l: l.account_id.id not in accounts_set_on_journal)`